firebase.go

```diff
@@ -99,6 +99,11 @@ func New(url string) *Firebase {
 		CheckRedirect: redirectPreserveHeaders,
 	}
 
+	return NewFromHttpClient(url, client)
+}
+
+// NewFromHttpClient creates a new Firebase reference with given http client
+func NewFromHttpClient(url string, client *http.Client) *Firebase {
```

Comment:
I'd prefer to get rid of the custom `http.Transport` and just have a single constructor that behaves similar to [go-github](https://github.com/google/go-github/blob/master/github/github.go#L128). If someone still wants requests to timeout after N seconds, they can pass a transport that does.

This would also end up resolving https://github.com/CloudCom/firego/issues/20